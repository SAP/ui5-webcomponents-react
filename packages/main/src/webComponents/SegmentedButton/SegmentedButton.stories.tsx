import type { Meta, StoryObj } from '@storybook/react';
import { SegmentedButtonItem } from '../SegmentedButtonItem/index.js';
import { SegmentedButton } from './index.js';

const meta = {
  title: 'Inputs / SegmentedButton',
  component: SegmentedButton,
  argTypes: {
    children: { control: { disable: true } }
  },
  args: {
    children: (
      <>
        <SegmentedButtonItem>Button 1</SegmentedButtonItem>
        <SegmentedButtonItem pressed>Button 2</SegmentedButtonItem>
        <SegmentedButtonItem>Button 3</SegmentedButtonItem>
      </>
    )
  }
} satisfies Meta<typeof SegmentedButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
