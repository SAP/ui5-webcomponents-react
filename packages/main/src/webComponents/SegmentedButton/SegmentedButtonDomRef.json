[
  {
    "name": "selectedItem",
    "readonly": "true",
    "visibility": "public",
    "type": "sap.ui.webc.main.ISegmentedButtonItem",
    "description": "Currently selected item."
  }
]
