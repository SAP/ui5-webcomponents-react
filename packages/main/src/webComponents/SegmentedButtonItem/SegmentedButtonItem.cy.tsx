import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { SegmentedButtonItem } from './index';

describe('SegmentedButtonItem', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<SegmentedButtonItem />);
    cy.get('[ui5-segmented-button-item]').should('exist');
  });

  cypressPassThroughTestsFactory(SegmentedButtonItem);
});
