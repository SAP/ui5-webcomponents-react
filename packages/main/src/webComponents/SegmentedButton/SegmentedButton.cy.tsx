import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { SegmentedButtonItem } from '../SegmentedButtonItem';
import { SegmentedButton } from './index';

describe('SegmentedButton', () => {
  it('Basic Test (generated)', () => {
    cy.mount(
      <SegmentedButton>
        <SegmentedButtonItem>Segment 1</SegmentedButtonItem>
        <SegmentedButtonItem>Segment 2</SegmentedButtonItem>
        <SegmentedButtonItem>Segment 3</SegmentedButtonItem>
      </SegmentedButton>
    );
    cy.get('[ui5-segmented-button]').should('exist');
  });

  cypressPassThroughTestsFactory(SegmentedButton);
});
