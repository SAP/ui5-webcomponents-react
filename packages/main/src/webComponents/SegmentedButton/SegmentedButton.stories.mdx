import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { SegmentedButton } from './index';
import { SegmentedButtonItem } from '../SegmentedButtonItem';
import { DocsHeader } from '@docs/DocsHeader';
import { Footer } from '@docs/Footer';

<Meta
  title="Inputs / SegmentedButton"
  component={SegmentedButton}
  subcomponents={{ SegmentedButtonItem }}
  argTypes={{
    children: { control: { disable: true } }
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return (
        <SegmentedButton {...args}>
          <SegmentedButtonItem>Button 1</SegmentedButtonItem>
          <SegmentedButtonItem pressed>Button 2</SegmentedButtonItem>
          <SegmentedButtonItem>Button 3</SegmentedButtonItem>
        </SegmentedButton>
      );
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

<Footer />
