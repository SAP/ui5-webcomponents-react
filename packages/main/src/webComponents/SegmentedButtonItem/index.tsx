import { ReactNode, MouseEventHandler } from 'react';
import { ButtonDesign } from '../../enums';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

import '@ui5/webcomponents/dist/SegmentedButtonItem.js';

interface SegmentedButtonItemAttributes {
  /**
   * Determines whether the component is displayed as pressed.
   */
  pressed?: boolean;
  /**
   * Sets the accessible aria name of the component.
   */
  accessibleName?: string;
  /**
   * Receives id(or many ids) of the elements that label the component.
   */
  accessibleNameRef?: string;
  /**
   * Defines whether the component is disabled. A disabled component can't be pressed or focused, and it is not in the tab chain.
   */
  disabled?: boolean;
  /**
   * Defines the icon to be displayed as graphical element within the component. The SAP-icons font provides numerous options.
   *
   * Example: See all the available icons in the <ui5-link target="_blank" href="https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.
   */
  icon?: string;
}

export interface SegmentedButtonItemDomRef extends SegmentedButtonItemAttributes, Ui5DomRef {}

export interface SegmentedButtonItemPropTypes extends SegmentedButtonItemAttributes, Omit<CommonProps, 'onClick'> {
  /**
   * Defines the text of the component.
   *
   * **Note:** Although this slot accepts HTML Elements, it is strongly recommended that you only use text in order to preserve the intended design.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Fired when the component is activated either with a mouse/tap or by using the Enter or Space key.
   *
   * **Note:** The event will not be fired if the `disabled` property is set to `true`.
   */
  onClick?: MouseEventHandler<HTMLElement>;
}

/**
 * Users can use the `SegmentedButtonItem` as part of a `SegmentedButton`.
 *
 * Clicking or tapping on a `SegmentedButtonItem` changes its state to `pressed`. The item returns to its initial state when the user clicks or taps on it again. By applying additional custom CSS-styling classes, apps can give a different style to any `SegmentedButtonItem`.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/SegmentedButtonItem" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const SegmentedButtonItem = withWebComponent<SegmentedButtonItemPropTypes, SegmentedButtonItemDomRef>(
  'ui5-segmented-button-item',
  ['accessibleName', 'accessibleNameRef', 'icon'],
  ['pressed', 'disabled'],
  [],
  ['click']
);

SegmentedButtonItem.displayName = 'SegmentedButtonItem';

export { SegmentedButtonItem };
