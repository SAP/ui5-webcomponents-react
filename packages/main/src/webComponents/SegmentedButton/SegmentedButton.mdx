import { ArgTypesWithNote, ControlsWithNote, DocsHeader, DomRefTable, Footer } from '@sb/components';
import SubcomponentsSection from '@sb/docs/SubcomponentsSection.md?raw';
import { Canvas, Markdown, Meta } from '@storybook/blocks';
import { SegmentedButtonItem } from '../SegmentedButtonItem';
import SegmentedButtonItemDescription from '../SegmentedButtonItem/SegmentedButtonItemDescription.md?raw';
import * as ComponentStories from './SegmentedButton.stories';
import SegmentedButtonDomRef from './SegmentedButtonDomRef.json';

<Meta of={ComponentStories} />

<DocsHeader since="0.9.0" />

<br />

## Example

<Canvas of={ComponentStories.Default} />

## Properties

<ControlsWithNote of={ComponentStories.Default} />

<DomRefTable rows={SegmentedButtonDomRef} />

<Markdown>{SubcomponentsSection}</Markdown>

## SegmentedButtonItem

<Markdown>{SegmentedButtonItemDescription}</Markdown>
<ArgTypesWithNote hideHTMLPropsNote of={SegmentedButtonItem} />

<Footer />
