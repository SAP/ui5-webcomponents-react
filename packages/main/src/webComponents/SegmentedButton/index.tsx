import { withWebComponent } from '@ui5/webcomponents-react/dist/withWebComponent';
import { CommonProps } from '@ui5/webcomponents-react/interfaces/CommonProps';
import { Ui5CustomEvent } from '@ui5/webcomponents-react/interfaces/Ui5CustomEvent';
import { Ui5DomRef } from '@ui5/webcomponents-react/interfaces/Ui5DomRef';
import { ReactNode } from 'react';

import '@ui5/webcomponents/dist/SegmentedButton.js';

interface SegmentedButtonAttributes {
  /**
   * Sets the accessible aria name of the component.
   */
  accessibleName?: string;
}

export interface SegmentedButtonDomRef extends SegmentedButtonAttributes, Ui5DomRef {
  /**
   * Currently selected item.
   */
  readonly selectedItem: ReactNode;
}

export interface SegmentedButtonPropTypes extends SegmentedButtonAttributes, CommonProps {
  /**
   * Defines the items of `SegmentedButton`.
   *
   * **Note:** Multiple items are allowed.
   *
   * **Note:** Use the `SegmentedButtonItem` for the intended design.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Fired when the selected item changes.
   */
  onSelectionChange?: (event: Ui5CustomEvent<HTMLButtonElement, { selectedItem: ReactNode }>) => void;
}

/**
 * The `SegmentedButton` shows a group of items. When the user clicks or taps one of the items, it stays in a pressed state. It automatically resizes the items to fit proportionally within the component. When no width is set, the component uses the available width.
 *
 * **Note:** There can be just one selected `item` at a time.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/SegmentedButton" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const SegmentedButton = withWebComponent<SegmentedButtonPropTypes, SegmentedButtonDomRef>(
  'ui5-segmented-button',
  ['accessibleName'],
  [],
  [],
  ['selection-change']
);

SegmentedButton.displayName = 'SegmentedButton';

export { SegmentedButton };
