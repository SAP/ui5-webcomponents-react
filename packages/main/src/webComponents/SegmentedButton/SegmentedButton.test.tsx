import { render } from '@shared/tests';
import { SegmentedButtonItem } from '@ui5/webcomponents-react/dist/SegmentedButtonItem';
import { SegmentedButton } from '@ui5/webcomponents-react/dist/SegmentedButton';
import React from 'react';

describe('SegmentedButton', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(
      <SegmentedButton>
        <SegmentedButtonItem>Segment 1</SegmentedButtonItem>
        <SegmentedButtonItem>Segment 2</SegmentedButtonItem>
        <SegmentedButtonItem>Segment 3</SegmentedButtonItem>
      </SegmentedButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
