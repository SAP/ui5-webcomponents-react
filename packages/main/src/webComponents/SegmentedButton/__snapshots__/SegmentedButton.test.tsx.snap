// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedButton Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-segmented-button
    ui5-segmented-button=""
  >
    <ui5-segmented-button-item
      ui5-segmented-button-item=""
    >
      Segment 1
    </ui5-segmented-button-item>
    <ui5-segmented-button-item
      ui5-segmented-button-item=""
    >
      Segment 2
    </ui5-segmented-button-item>
    <ui5-segmented-button-item
      ui5-segmented-button-item=""
    >
      Segment 3
    </ui5-segmented-button-item>
  </ui5-segmented-button>
</DocumentFragment>
`;
